orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/appengine/AbstractWaitForAppEngineServerGroupStopStartTask.groovy

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.providers.appengine
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.frigga.Names
+import com.netflix.spinnaker.orca.DefaultTaskResult
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.RetryableTask
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.clouddriver.OortService
+import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import com.netflix.spinnaker.orca.retrofit.exceptions.RetrofitExceptionHandler
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import retrofit.RetrofitError
+
+@Slf4j
+abstract class AbstractWaitForAppEngineServerGroupStopStartTask extends AbstractCloudProviderAwareTask implements RetryableTask {
+  long backoffPeriod = 10000
+  long timeout = 1800000
+
+  @Autowired
+  OortService oortService
+
+  @Autowired
+  ObjectMapper objectMapper
+
+  abstract boolean isStart()
+
+  @Override
+  TaskResult execute(Stage stage) {
+    String cloudProvider = getCloudProvider(stage)
+    String account = getCredentials(stage)
+    String serverGroupName = (stage.context.serverGroupName ?: stage.context.asgName) as String
+    Names names = Names.parseName(serverGroupName)
+    try {
+      def response = oortService.getCluster(names.app, account, names.cluster, cloudProvider)
+
+      if (response.status != 200) {
```

Comment:
I see this in a few places (e.g., `WaitForDestroyedServerGroupTask` or `AbstractInstancesCheckTask`). I'm assuming that Clouddriver can return a `2xx` response such that we'll want to keep the task status as `RUNNING` but not want to check the response body for pass/fail conditions.